test/test_resize.cpp

```diff
@@ -2252,4 +2252,197 @@ TEST_F(NVFuserTest, SliceVectorization) {
+      1,
+      1, // dim=2 pad with zeros on either side
+      -1,
+      -1, // dim=1 pad to broadcast of second element
+      // dim=0 is implicit 0, 0
+  };
+  std::vector<IterType> expected_itertypes = {
+      IterType::Iteration,
+      IterType::Broadcast,
+      IterType::Iteration,
+      IterType::Broadcast,
+      IterType::Iteration,
+  };
+
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  auto tv0 = makeConcreteTensor(t0_size);
+  fusion->addInput(tv0);
+  auto tv1 = makeConcreteTensor(t1_size);
+  fusion->addInput(tv1);
+
+  std::vector<Val*> pad_width_vals;
+  pad_width_vals.reserve(pad_widths.size());
+  for (auto w : pad_widths) {
+    pad_width_vals.push_back(IrBuilder::create<Scalar>(w));
+  }
+
+  auto tv2 = pad(tv0, pad_width_vals);
+  auto tv3 = mul(tv1, tv2);
+  fusion->addOutput(tv3);
+
+  EXPECT_FALSE(fusion->hasDynamicTransform());
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+
+  auto t0 = at::randn(t0_size, options);
+  auto t1 = at::randn(t1_size, options);
+  std::vector<c10::IValue> aten_inputs({t0, t1});
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto cg_outputs = executor_cache.runFusionWithInputs(aten_inputs);
+
+  auto runtime = executor_cache.getMostRecentKernelRuntime();
+  auto concretized_fusion = runtime->fusionSegments()->completeFusion();
+
+  auto conc_t2 = concretized_fusion->outputs()[0]
+                     ->definition()
+                     ->inputs()[1]
+                     ->as<TensorView>();
+  for (auto i : c10::irange(expected_itertypes.size())) {
+    EXPECT_EQ(conc_t2->axis(i)->getIterType(), expected_itertypes.at(i));
+  }
+
+  auto t2_padded = at::pad(t0, pad_widths);
+  auto ref_t2 = t1 * t2_padded;
+
+  testValidate(
+      concretized_fusion,
+      cg_outputs,
+      aten_inputs,
+      {ref_t2},
+      __LINE__,
+      __FILE__);
+}
+
+// Concretize a symbolic pad that results in a broadcast (dynamic pads)
+TEST_F(NVFuserTest, ResizePadToBroadcastDynamic_CUDA) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  auto tv0 = makeSymbolicTensor(5);
+  fusion->addInput(tv0);
+  auto tv1 = makeSymbolicTensor(5);
+  fusion->addInput(tv1);
+
+  // Note there are only 8 input scalars for 5D input. Implicit no-pad of dim 0
+  std::vector<int64_t> pad_widths = {
+      0,
+      -1, // dim=4 trim last element
+      0,
+      -4, // dim=3 pad to broadcast of first element
+      1,
+      1, // dim=2 pad with zeros on either side
+      -1,
+      -1, // dim=1 pad to broadcast of second element
+      // dim=0 is implicit 0, 0
+  };
+  std::vector<Val*> pad_width_vals;
+  pad_width_vals.reserve(pad_widths.size());
+  for ([[maybe_unused]] auto _ : pad_widths) {
+    auto w_val = IrBuilder::create<Scalar>(DataType::Int);
+    fusion->addInput(w_val);
+    pad_width_vals.push_back(w_val);
+  }
+
+  auto tv2 = pad(tv0, pad_width_vals);
+  auto tv3 = mul(tv1, tv2);
```

Comment:
This test currently fails. The Fusion is concretized properly:
```
Inputs:
  T0_g[ iS0{i0}, iS1{i2}, iS2{i3}, iS3{i4}, iS4{i5} ], float
  T1_g[ iS5{i6}, iS6{i7}, iS7{i8}, iS8{i9}, iS9{i10} ], float
  i11, int64_t
  i12, int64_t
  i13, int64_t
  i14, int64_t
  i15, int64_t
  i16, int64_t
  i17, int64_t
  i18, int64_t
Outputs:
  T3_g[ iS19{i6}, iS20{i7}, iS21{i8}, iS22{i9}, iS23{i10} ], float                                                               
%kernel_math {
T2_l[ iS10{i0}, bS24{( ( i2 + i17 ) + i18 )}rf, iS25{( ( i3 + i15 ) + i16 )}rf, bS26{( ( i4 + i13 ) + i14 )}rf, iS27{( ( i5 + i11
 ) + i12 )}rf ]
   = pad( T0_g[ iS0{i0}, iS1{i2}, iS2{i3}, iS3{i4}, iS4{i5} ], {0, 0, i17, i18, i15, i16, i13, i14, i11, i12} )
T3_g[ iS19{i6}, iS20{i7}, iS21{i8}, iS22{i9}, iS23{i10} ]
   = T1_g[ iS5{i6}, iS6{i7}, iS7{i8}, iS8{i9}, iS9{i10} ]
   * T2_l[ iS10{i0}, bS24{( ( i2 + i17 ) + i18 )}rf, iS25{( ( i3 + i15 ) + i16 )}rf, bS26{( ( i4 + i13 ) + i14 )}rf, iS27{( ( i5
+ i11 ) + i12 )}rf ];
}
```
but then it is segmented at `tv2` (pointwise rejects with `cannot find reference tensor`). I think the pointwise scheduler should be able to schedule this, but the segmented Fusion also dies, since we cannot bind the input shape to `tv2` since it has a definition.

I think we may just need to rewrite the extent of the output as `oneVal()` when concretizing resizes as broadcasts.

Reply:
This was fixed in #630 and these tests are now passing after merging from main.